tests/manage/z_cluster/test_must_gather_options.py

```diff
@@ -0,0 +1,89 @@
+        assert "KUBECONFIG" in os.environ or os.path.exists(
+            os.path.expanduser("~/.kube/config")
+        ), "Cannot find $KUBECONFIG or ~/.kube/config; skipping log collection"
+        self.temp_folder = tempfile.mkdtemp()
+
+    def teardown(self):
+        logger.info(f"Delete must gather folder {self.temp_folder}")
+        if os.path.exists(path=self.temp_folder):
+            exec_cmd(cmd=f"rm -rf {self.temp_folder}")
+
+    @tier2
+    @pytest.mark.parametrize(
+        argnames=["expected_result", "cmd_param", "timeout"],
+        argvalues=[
+            pytest.param(*["Pass", "request-timeout", "1200s"]),
+            pytest.param(*["Pass", "request-timeout", "20m"]),
+            pytest.param(*["Pass", "request-timeout", "1h"]),
+            pytest.param(*["Pass", "timeout", "1200"]),
+            pytest.param(*["Fail", "timeout", "-1200"]),
+            pytest.param(*["Fail", "timeout", "-1200s"]),
+            pytest.param(*["Fail", "timeout", "-20m"]),
+            pytest.param(*["Fail", "timeout", "-1h"]),
+        ],
+    )
+    @pytest.mark.skipif(
+        float(config.ENV_DATA["ocs_version"]) not in GATHER_COMMANDS_VERSION,
+        reason=(
+            "Skipping must_gather test, because there is not data for this version"
+        ),
+    )
+    def test_must_gather_timeout(self, expected_result, cmd_param, timeout):
+        """
+        Test must-gather timeout parameter
+        """
+        logger.info(
+            f"Running must-gather command with '{cmd_param}={timeout}'. Operation should {expected_result}."
+        )
+        latest_tag = config.REPORTING.get(
+            "ocs_must_gather_latest_tag",
+            config.REPORTING.get(
+                "default_ocs_must_gather_latest_tag",
+                config.DEPLOYMENT["default_latest_tag"],
+            ),
+        )
+        ocs_must_gather_image = config.REPORTING["ocs_must_gather_image"]
+        image = f"{ocs_must_gather_image}:{latest_tag}"
+        if config.DEPLOYMENT.get("disconnected"):
+            image = mirror_image(image)
+
+        logger.info(f"Must gather image: {image} will be used.")
+        # create_directory_path(self.temp_folder)
+        cmd = f"adm must-gather --image={image} --dest-dir={self.temp_folder} --{cmd_param}={timeout}"
+
+        occli = OCP()
+        try:
+            occli.exec_oc_cmd(cmd, out_yaml_format=False, timeout=1200)
+        except CommandFailed as ex:
+            # assertion when the must-gather failed but should pass
+            assert (
+                not expected_result == "Pass"
+            ), f"Must gather failed when it should pass! Error: {ex}"
+            # must-gather failed as expected
+            logger.info(f"Failed during must gather as expected. Error: {ex}")
+        except TimeoutExpired as ex:
+            logger.error(
+                f"Timeout {timeout}s for must-gather reached, command"
+                f" exited with error: {ex}"
+            )
```

Comment:
Maybe we can add another log message at the end of the test in case we expect it to pass - so we should know the test finished successfully.

Reply:
Removed